lib/engines/hckinstall/hckinstall.rb

```diff
@@ -282,13 +280,14 @@ def prepare_setup_scripts_config
       }
 
       unless @kit_info['download_url'].nil?
-        download_kit_installer(@kit_info['download_url'],
-                               "#{kit_type}#{kit_version}", @hck_setup_scripts_path)
+        @kit_path = download_kit_installer(@kit_info['download_url'],
+                                           "#{kit_type}#{kit_version}", @hck_setup_scripts_path, iso: @kit_info['iso'])
       end
 
       installers = [
         "#{@hck_setup_scripts_path}/Kits/#{kit_type}#{kit_version}Setup.exe",
-        "#{@hck_setup_scripts_path}/Kits/#{kit_type}#{kit_version}/#{kit_type}Setup.exe"
+        "#{@hck_setup_scripts_path}/Kits/#{kit_type}#{kit_version}/#{kit_type}Setup.exe",
+        "#{@hck_setup_scripts_path}/Kits/#{kit_type}#{kit_version}Setup.iso"
```

Comment:
It's not right to add `"#{@hck_setup_scripts_path}/Kits/#{kit_type}#{kit_version}Setup.iso"` unconditionally. It will result in an incorrect behavior if `@kit_info['iso']` is true and `@kit_info['download_url']` is `nil`.

Reply:
I expect `@kit_info['iso'] is true` and `@kit_info['download_url'] is nil` in production with Insider Preview testing. Microsoft provides temporary URLs, but we can rerun the installation several times due to a problem with Insider Preview Windows / HLK. 

I expect that the HLKInstall engine will see ISO and be as happy as currently with EXE.
```
unless @kit_info['download_url'].nil?
        download_kit_installer(@kit_info['download_url'],
                               "#{kit_type}#{kit_version}", @hck_setup_scripts_path)
      end
````

We download the installer only if download_url is present, but if I manually extract HLK ISO and `"#{@hck_setup_scripts_path}/Kits/#{kit_type}#{kit_version}/#{kit_type}Setup.exe"` will be present, then HLKInstall engine will continue work. With ISO I expect the same behavior without download_url.